src/v/kafka/server/group_manager.cc

```diff
@@ -746,6 +762,75 @@ ss::future<> group_manager::do_recover_group(
     co_return;
 }
 
+ss::future<> group_manager::write_version_fence(
+  model::term_id term, ss::lw_shared_ptr<attached_partition> p) {
+    // how long to delay retrying a fence write if an error occurs
+    constexpr auto fence_write_retry_delay = 10s;
+
+    co_await _feature_table.local().await_feature(
+      features::feature::group_offset_retention, p->as);
```

Comment:
nit: if you rebase you could try using the `await_feature_then()` method to encapsulate some of the failure handling (may not be worth it, considering this uses the partition's abort source rather than the feature table's)